File: Examples/Sample-ObjC/Sample-ObjC/ViewController.m
//
//  ViewController.m
//  Sample-ObjC
//
//  Created by Vishal Nandoriya on 27/07/22.
//

#import "ViewController.h"
#import <SalesmateChatSDK/SalesmateChatSDK.h>

@interface ViewController ()

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view.

    [self setVerifiedId];
}

- (void)setVerifiedId {
    if ([SalesmateChat getVisitorId].length != 0) {
        [SalesmateChat setVerifiedID: [SalesmateChat getVisitorId]];
    }
}

//MARK: - IBActions
- (IBAction)presentMessenger:(id)sender {
    [SalesmateChat presentMessengerFrom: self];
}

- (IBAction)logEvent:(id)sender {
    NSString *eventName = @"Testing Event";
    NSString *key = @"testEventKey";
    NSString *value = @"testEventValue";
    
    [SalesmateChat logEventWithEventName: eventName withData: @{key: value}];
    
    [self showAlertWithTitle: @"Event Queued for delivery" withMessage: @""];

}

- (IBAction)loginPressed:(id)sender {
    [self userLogin];
}

- (IBAction)updatePressed:(id)sender {
    
    /// Update
    /*[SalesmateChat updateWithUserId:"" email:"" firstName:"" lastName:"" completion:^(NSString * _Nullable, NSError * _Nullable) { }]*/
}

- (IBAction)logout:(id)sender {
    [SalesmateChat logout];
    [self showAlertWithTitle: @"Salesmate Chat" withMessage: @"Logout Success"];
}

- (void)userLogin {
    NSString  *userId = [self randomFakeUserId];
    NSString *email = [self randomFakeEmail];
    NSString *firstName = [self randomFakeFirstName];
    NSString *lastName = [self randomFakeLastName];
    
    [SalesmateChat loginWithUserId: userId email: email firstName: firstName lastName: lastName completion: ^(NSString *success, NSError *error) {
        if (error == nil) {
            __weak typeof(self) weakSelf = self;
            dispatch_async(dispatch_get_main_queue(), ^{
                [weakSelf showAlertWithTitle: @"Salesmate Chat" withMessage: @"Login Success"];
            });
        }
    }];

}

- (NSString *)randomFakeUserId {
    NSArray *userIdList = @[@"Henry123", @"William234", @"Geoffrey345", @"Jim456", @"Yvonne567", @"Jamie678", @"Leticia789", @"Priscilla890", @"Sidney012", @"Nancy109", @"Edmund298", @"Bill376", @"Megan037"];
    
    return  [userIdList objectAtIndex: arc4random_uniform(12)];
}

- (NSString *)randomFakeEmail {
    NSArray *emailList = @[@"Henry@test.com", @"William@test.com", @"Geoffrey@test.com", @"Jim@test.com", @"Yvonne@test.com", @"Jamie@test.com", @"Leticia@test.com", @"Priscilla@test.com", @"Sidney@test.com", @"Nancy@test.com", @"Edmund@test.com", @"Bill@test.com", @"Megan@test.com"];
    
    return  [emailList objectAtIndex: arc4random_uniform(12)];

}

- (NSString *)randomFakeFirstName {
    NSArray *firstNameList = @[@"Henry", @"William", @"Geoffrey", @"Jim", @"Yvonne", @"Jamie", @"Leticia", @"Priscilla", @"Sidney", @"Nancy", @"Edmund", @"Bill", @"Megan"];
    
    return  [firstNameList objectAtIndex: arc4random_uniform(12)];

}

- (NSString *)randomFakeLastName {
    NSArray *lastNameList = @[@"Pearson", @"Adams", @"Cole", @"Francis", @"Andrews", @"Casey", @"Gross", @"Lane", @"Thomas", @"Patrick", @"Strickland", @"Nicolas", @"Freeman"];
    
    return  [lastNameList objectAtIndex: arc4random_uniform(12)];

}

- (void)showAlertWithTitle:(NSString *)title withMessage: (NSString *)message {
    
    UIAlertController* alert = [UIAlertController alertControllerWithTitle: title
                               message: message
                               preferredStyle:UIAlertControllerStyleAlert];

    UIAlertAction* defaultAction = [UIAlertAction actionWithTitle:@"OK" style:UIAlertActionStyleDefault
                                   handler:^(UIAlertAction * action) {}];

    [alert addAction:defaultAction];
    [self presentViewController:alert animated:YES completion:nil];
}

@end

File: Examples/Sample-Swift/Sample-Swift/ViewController.swift
//
//  ViewController.swift
//  Sample-Swift
//
//  Created by Vishal Nandoriya on 27/07/22.
//

import UIKit
import SalesmateChatSDK

class ViewController: UIViewController {

    //MARK: - View Life Cycle
    override func viewDidLoad() {
        super.viewDidLoad()
        
        // Set Visitor Id
        setVerifiedId()
    }

    func setVerifiedId() {
        guard !SalesmateChat.getVisitorId().isEmpty else {
            return
        }
        SalesmateChat.setVerifiedID(SalesmateChat.getVisitorId())
    }
    
    //MARK: - IBActions
    @IBAction func presentMessenger(_ sender: UIButton) {
        SalesmateChat.presentMessenger(from: self)
    }

    @IBAction func logEvent(_ sender: Any) {
        
        let eventName = "Testing Event"
        let key = "testEventKey"
        let value = "testEventValue"
        
        SalesmateChat.logEventWith(eventName: eventName, withData: [key:value]);
        
        
        self.showAlertWithTitle(title: "Event Queued for delivery", andMessage: "");
    }

    @IBAction func loginPressed(_ sender: Any) {
        userLogin()
    }
    
    @IBAction func updatePressed(_ sender: UIButton) {
        
        /// Update
        // SalesmateChat.update(userId: "", email: "", firstName: "", lastName: "", completion: nil)
    }

    @IBAction func logout(_ sender: Any) {
        SalesmateChat.logout()
        self.showAlertWithTitle(title: "Salesmate Chat", andMessage: "Logout Success")
    }
}

extension ViewController {
    
    func userLogin() {
        let userId = randomFakeUserId()
        let email = randomFakeEmail()
        let firstName = randomFakeFirstName()
        let lastName = randomFakeLastName()

        SalesmateChat.loginWith(userId: userId, email: email, firstName: firstName, lastName: lastName, completion: { (success, error)  in
            if error == nil {
                DispatchQueue.main.async {
                    self.showAlertWithTitle(title: "Salesmate Chat", andMessage: "Login Success")
                }
            }
        })
    }

    func randomFakeUserId() -> String {
        let userIdList = ["Henry123", "William234", "Geoffrey345", "Jim456", "Yvonne567", "Jamie678", "Leticia789", "Priscilla890", "Sidney012", "Nancy109", "Edmund298", "Bill376", "Megan037"]
        return userIdList.randomElement()!
    }

    func randomFakeEmail() -> String {
        let emailList = ["Henry@test.com", "William@test.com", "Geoffrey@test.com", "Jim@test.com", "Yvonne@test.com", "Jamie@test.com", "Leticia@test.com", "Priscilla@test.com", "Sidney@test.com", "Nancy@test.com", "Edmund@test.com", "Bill@test.com", "Megan@test.com"]
        return emailList.randomElement()!
    }

    func randomFakeFirstName() -> String {
        let firstNameList = ["Henry", "William", "Geoffrey", "Jim", "Yvonne", "Jamie", "Leticia", "Priscilla", "Sidney", "Nancy", "Edmund", "Bill", "Megan"]
        return firstNameList.randomElement()!
    }
    
    func randomFakeLastName() -> String {
        let lastNameList = ["Pearson", "Adams", "Cole", "Francis", "Andrews", "Casey", "Gross", "Lane", "Thomas", "Patrick", "Strickland", "Nicolas", "Freeman"]
        return lastNameList.randomElement()!
    }

    func showAlertWithTitle(title:String, andMessage message:String){
        let alertVC = UIAlertController(title: title, message: message, preferredStyle: .alert);
        let cancelAction = UIAlertAction(title: "OK", style: .cancel, handler: nil);
        alertVC.addAction(cancelAction);
        self.present(alertVC, animated: true, completion: nil);
    }

}
